src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -971,7 +848,8 @@ private void FastKill()
 
             // Setting the event should be the last thing we do.
             // Do nothijng after that!
-            siloTerminatedEvent.Set();  
+            siloTerminatedTask.SetResult(0);
+            siloTerminatedEvent.Set();
```

Comment:
I can't see how. I thought about uniting them. I think we should obsolete/eliminate the WaitHandle instead, it was always annoying to use.

`SiloHost` can use the task in `WaitForOrleansSiloShutdown` (and we can hopefully obsolete `SiloHost`, too)